### ISSUE TITLE
Clarify MacOS Brew Instructions for extra files

### ISSUE THREAD

**Issue by jessebett** [alacritty/alacritty#2653]
The installation [instructions for MacOS in the README](https://github.com/jwilm/alacritty#macos) use homebrew. Following the command it gives the advice to set up additional things.
>Once the cask is installed, it is recommended to setup the manual page, shell completions, and terminfo definitions.

However, those instructions provide scripts which reference local folders like `extra`, e.g. 
```
sudo mkdir -p /usr/local/share/man/man1
gzip -c extra/alacritty.man | sudo tee /usr/local/share/man/man1/alacritty.1.gz > /dev/null
```

Following the macOS install procedure, it is not clear where this folders are.

**Comment by chrisduerr** [comment]
As far as I know, they aren't anywhere. This is a brew issue not an Alacritty one. You have to clone the repository and copy it yourself as far as I know, though I'm not super into the odd macOS package management system.

**Comment by jessebett** [comment]
Can the README be updated to something which instructs you to clone the repo :

>Once the cask is installed, it is recommended to clone this repo to setup the manual page, shell completions, and terminfo definitions.

**Comment by chrisduerr** [comment]
I'd think that through the links to the INSTALL.md, most of that should be explained in the document already, however since it links to specific sections I can see how that might confuse some users who aren't super familiar with the process. So I'd be fine with pointing out explicitly that these instructions require a copy of Alacritty's repo to be completed (though not necessarily require a full clone, but people that care about that should know that).

**Comment by hsxfjames** [comment]
I see the files like `alacritty.1.gz` and `alacritty.info` can be download from release page of this repo. I think https://github.com/Homebrew/homebrew-cask/blob/master/Casks/alacritty.rb may have to automatically copy and link these files, so users don't need to do this by themself.

**Comment by hsxfjames** [comment]
I see the files like `alacritty.1.gz` and `alacritty.info` can be downloaded from release page of this repo. I think https://github.com/Homebrew/homebrew-cask/blob/master/Casks/alacritty.rb might be able to automatically copy and link these files, so users don't need to do this by themself. A cask could execute system command in postflight section.

**Comment by chrisduerr** [comment]
> I think https://github.com/Homebrew/homebrew-cask/blob/master/Casks/alacritty.rb might be able to automatically copy and link these files, so users don't need to do this by themself, since a cask could execute system command in postflight section.

The brew package is maintained externally, so proposals about a change to it need to be addressed to its maintainers.

**Comment by 0az** [comment]
Would it be possible to bundle the relevant extras (completion, manpage, terminfo) with the final App? Based on my (*very* cursory) search of the Homebrew Cask docs, there's no good way to download extras. On the other hand, there is express support for linking manpages and other artifacts. Completion's handled manually by users, but bundling it would at least eliminate an extra `git clone` or `wget`.

**Comment by chrisduerr** [comment]
Alacritty does not manage any packages. If you have problems with any of them, you'll have to contact the third party to fix it.

**Comment by 0az** [comment]
Let me rephrase: would it be possible to copy `extras/completion`, the terminfo, and the manpage into the MacOS app?

**Comment by chrisduerr** [comment]
They're already available for each release on the GitHub download page, so shouldn't that suffice?

Just having it inside the MacOS app doesn't help unless the user actual copies it to the correct location and that's probably just as easy from separate files I'd assume?

**Comment by 0az** [comment]
I don't believe that Homebrew in particular has a good way to download additional files: afaik, it assumes everything's in one archive with a fixed hash, which in this case is the standard disk image.

And even in a Homebrew-less world, one download is more convenient, regardless of it's somewhere below `.app`. At the very least, it'd make the install instructions clearer in the common case.

**Comment by chrisduerr** [comment]
I'm not entirely familiar with the sad state of software management on macOS. However to me it seems like just bundling everything in one big Alacritty.app would needlessly bloat it in a lot of cases.

However if that's the best way to provide these files on macOS, I'll happily accept contributions that implement this. Though I'd be curious how other applications handle this.

**Comment by 0az** [comment]
Anyway, I'll see about sending in a PR for this sometime over the next week. Shouldn't be that bad, based on my cursory look at the Makefile earlier today.

Few casks have terminal binaries, and fewer bother including manpages (most casks are GUI). This is mostly because the only pure-cli tools with a cask are not open source. However, those that do have a manpage include it with the download.

<details>
<summary>All of the casks with manpages.</summary>

```sh
$ rg -coF 'manpage' -g '*.rb' (brew --prefix)/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/jtool.rb:1
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/diffmerge.rb:1
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/manopen.rb:1
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/hfsleuth.rb:1
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/procexp.rb:1
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/jad.rb:1
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/mpv.rb:1
```

</details>

**Comment by nixpulvis** [comment]
I suppose in theory the cask could include an installer, and that might be the best way to handle this. It'd be nice to hear what someone from the cask has to say: https://github.com/Homebrew/homebrew-cask/blob/master/Casks/alacritty.rb

**Comment by deekue** [comment]
[potential cask change](https://github.com/deekue/homebrew-cask/commit/88adb3b1198db6b62bd43fa981814ffc1b4e05b0) to support the extra files

if the Makefile copied alacritty.man to the bundle as alacritty.1 then the cask file could use the 'manpage' command instead of 'system_command'

**Comment by nixpulvis** [comment]
@deekue now that this should be resolved in `master` I'm looking at the cask change you linked (out of curiosity, as I'm not the one who maintains the actual cask formula). First of all, I'm not 100% sure what command you mean by `manpage`, since a quick look didn't find anything by that name. I do see `man.` commands, but they didn't seem to work here. Do you have a working update to this formula you can share now?

**Comment by chrisduerr** [comment]
I'd assume that instead of a `system_command` a simple `FileUtils.mv` should do the job now. The format's already good, only need to install it now.

**Comment by nixpulvis** [comment]
Ah interesting I was assuming macOS needed to expand the gzip because almost all the manuals are uncompressed, however you're right, this should not be necessary!

**Comment by deekue** [comment]
the [manpage](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/all_stanzas.md#at-least-one-artifact-stanza-is-also-required) artifact stanza seems to assume extra files will be in the root of the .dmg ([examples](https://github.com/Homebrew/homebrew-cask/search?q=manpage&unscoped_q=manpage)).  
since Alacritty will have all it's extras in the app dir, I just used FileUtils.ln_sf 
I can send a PR once the next release goes out with the extras in it.
